src/main/java/seedu/address/logic/commands/BookCommand.java

```diff
@@ -71,4 +81,38 @@ public boolean equals(Object other) {
                 && targetIndex.equals(((BookCommand) other).targetIndex)
                 && appointment.equals(((BookCommand) other).appointment)); // state check
     }
+
+    private static class PersonBookDescriptor {
+        private final Name name;
+        private final Phone phone;
+        private final Email email;
+        private final Address address;
+        private final List<Appointment> appointments;
+        private final Set<Tag> tags;
+
+        private PersonBookDescriptor(Person toCopy) {
+            name = toCopy.getName();
+            phone = toCopy.getPhone();
+            email = toCopy.getEmail();
+            address = toCopy.getAddress();
+            appointments = new ArrayList<>(toCopy.getAppointments());
+            tags = toCopy.getTags();
+        }
+
+        private boolean hasSameAppointment(Appointment appointment) {
+            return appointments.stream().anyMatch(x -> x.isSameTime(appointment));
+        }
+
+        private void bookAppointment(Appointment appointment) throws CommandException {
+            if (hasSameAppointment(appointment)) {
+                throw new CommandException(MESSAGE_DUPLICATE_APPOINTMENT);
+            }
+            appointments.add(appointment);
+            appointments.sort(Comparator.comparing(Appointment::getDateTime));
+        }
```

Comment:
Perhaps it may be better if these commands are abstracted out? It feels slightly out of place here. I think that we can maybe extract these methods out to an AppointmentList class in the future.